src/core/amount.js

```diff
@@ -729,7 +703,8 @@ Amount.prototype.parse_native = function(j) {
 // Requires _currency to be set!
 Amount.prototype.parse_value = function(j) {
   this._is_native = false;
-  this._set_value(new BigNumber(j), BigNumber.ROUND_DOWN);
+  const newValue = new IOUValue(j).toPrecision(16, Value.getBNRoundDown());
```

Comment:
`toPrecision` sounds like it should return a string since that's what bignumber.js toPrecision returns. Maybe a better option would be to pass the rounding mode to the IOUValue constructor and call canonicalize from the constructor.